nixos/modules/services/web-servers/lighttpd/inginious.nix

```diff
@@ -0,0 +1,257 @@
+      example = ''[ "john" "pepe" "emilia" ]'';
+      description = ''List of user logins allowed to administrate the whole server.'';
+    };
+
+    containers = mkOption {
+      type = types.attrsOf types.str;
+      default = {
+          default = "ingi/inginious-c-default";
+      };
+      example = ''
+        {
+          default = "ingi/inginious-c-default";
+          sekexe  = "ingi/inginious-c-sekexe";
+          java    = "ingi/inginious-c-java";
+          oz      = "ingi/inginious-c-oz";
+          pythia1compat = "ingi/inginious-c-pythia1compat";
+        }
+      '';
+      description = ''
+        An attrset describing the required containers
+        These containers will be available in INGInious using their short name (key)
+        and will be automatically downloaded before INGInious starts.
+      '';
+    };
+
+    hostPattern = mkOption {
+      type = types.str;
+      default = "^inginious.";
+      example = "^inginious.mydomain.xyz$";
+      description = ''
+        The domain that serves INGInious.
+        INGInious uses absolute paths which makes it difficult to relocate in its own subdir.
+        The default configuration will serve INGInious when the server is accessed with a hostname starting with "inginious.".
+        If left blank, INGInious will take the precedence over all the other lighttpd sites, which is probably not what you want.
+      '';
+    };
+
+    local = mkOption {
+      type = types.bool;
+      default = true;
+      description = ''
+        Set to false if you want to use remote grading agents.
+        Remember to provide a list of agents (see `remoteAgents`) in that case.
+      '';
+    };
+
+    remoteAgents = mkOption {
+      type = types.listOf (types.attrsOf types.str);
+      default = [];
+      example = [ { host = "145.33.17.89"; port = "1345"; } ];
+      description = ''A list of remote agents.'';
+    };
+  };
+
+  config = mkIf cfg.enable (
+    mkMerge [
+      # For a local install, we need docker.
+      (mkIf cfg.local {
+        virtualisation.docker = {
+          enable = true;
+          # We need docker to listen on port 2375.
+          extraOptions = "-H tcp://localhost:2375 -H unix:///var/run/docker.sock";
+          storageDriver = mkDefault "overlay";
+          socketActivation = false;
+        };
+
+        users.extraUsers."lighttpd".extraGroups = [ "docker" ];
+      })
+
+      # Common
+      {
+        # To access inginous tools (like inginious-test-task)
+        environment.systemPackages = [ inginious ];
+
+        services.mongodb.enable = true;
+
+        services.lighttpd.enable = true;
+        services.lighttpd.enableModules = [ "mod_access" "mod_alias" "mod_fastcgi" "mod_redirect" "mod_rewrite" ];
+        services.lighttpd.extraConfig = ''
+          $HTTP["host"] =~ "${cfg.hostPattern}" {
+            fastcgi.server = ( "/${execName}" =>
+              ((
+                "socket" => "/tmp/fastcgi.socket",
```

Comment:
This is the example given by web.py and lighttpd. I see no reason why it could not be moved, but I am not sure that changing the value here is enough. I need investigations.